src/target/riscv/riscv-013.c

```diff
@@ -390,18 +390,29 @@ static void dump_field(struct target *target, int idle, const struct scan_field
 	unsigned int in_address = in >> DTM_DMI_ADDRESS_OFFSET;
 
 	log_printf_lf(LOG_LVL_DEBUG,
-			__FILE__, __LINE__, "scan",
+			__FILE__, __LINE__, __PRETTY_FUNCTION__,
 			"%db %s %08x @%02x -> %s %08x @%02x; %di",
 			field->num_bits, op_string[out_op], out_data, out_address,
 			status_string[in_op], in_data, in_address, idle);
 
-	char out_text[decode_dmi(target, NULL, out_address, out_data) + 1];
-	unsigned int out_len = decode_dmi(target, out_text, out_address, out_data);
-	char in_text[decode_dmi(target, NULL, in_address, in_data) + 1];
-	unsigned int in_len = decode_dmi(target, in_text, in_address, in_data);
-	if (in_text[0] || out_text[0]) {
-		log_printf_lf(LOG_LVL_DEBUG, __FILE__, __LINE__, "scan", "%.*s -> %.*s",
-				out_len, out_text, in_len, in_text);
+	const bool decode_out = out_op == DTM_DMI_OP_WRITE;
+	char out_decoded[decode_dmi(target, NULL, out_address, out_data) + 1];
+	const char *out_text = "";
+	if (decode_out)
+		out_text = decode_dmi(target, out_decoded, out_address, out_data)
+			? out_decoded
+			: "<no decoding available>";
+
+	char in_decoded[decode_dmi(target, NULL, in_address, in_data) + 1];
+	const char *in_text = "";
+	if (decode_in)
+		in_text = decode_dmi(target, in_decoded, in_address, in_data)
+			? in_decoded
+			: "<no decoding available>";
+
+	if (decode_out || decode_in) {
+		log_printf_lf(LOG_LVL_DEBUG, __FILE__, __LINE__, __PRETTY_FUNCTION__,
+				"%s -> %s", out_text, in_text);
```

Comment:
I would recommend to print the decoded values onto separate lines. And also to indent them (e.g. by 2 spaces) so that in the log they appear visually related to the preceding raw DMI print. Something like:

```suggestion
	if (decode_out) {
		log_printf_lf(LOG_LVL_DEBUG, __FILE__, __LINE__, __PRETTY_FUNCTION__,
				"  writing: %s", out_text);
		// or "out:" instead of "writing:"
	}
	if (decode_in) {
		log_printf_lf(LOG_LVL_DEBUG, __FILE__, __LINE__, __PRETTY_FUNCTION__,
				"  read: %s", in_text);
		// or "in:" instead of "read:"
	}	
```

Reasons:
- The decoded text produces very long log lines, especially when both "in" and "out" would be printed.
- When only one of the directions is decoded, it would be easier to tell if it is "in" or "out" by reading the prefix than checking whether the text is on the left (or on the right) of the `->` sign.

Reply:
Addressed.

Reply:
Thanks.